packages/starknet-snap/src/utils/starknetUtils.test.ts

```diff
@@ -0,0 +1,132 @@
+    prepareMockAccount(account, state);
+    const accountDeployedSpy = jest.spyOn(starknetUtils, 'isAccountDeployed');
+    accountDeployedSpy.mockResolvedValue(deployed);
+
+    const estimateFeeResp = {
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      overall_fee: BigInt('0xa'),
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      gas_consumed: BigInt('0x0'),
+      suggestedMaxFee: BigInt('0xc'),
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      gas_price: BigInt('0x0'),
+    } as EstimateFee;
+    const estimateBulkFeeSpy = jest.spyOn(starknetUtils, 'estimateFeeBulk');
+    estimateBulkFeeSpy.mockResolvedValue([estimateFeeResp]);
+
+    return {
+      account,
+      accountDeployedSpy,
+      estimateBulkFeeSpy,
+    };
+  };
+
+  it.each([constants.TRANSACTION_VERSION.V2, constants.TRANSACTION_VERSION.V3])(
+    'estimate fees for transaction version %s',
+    async (transactionVersion) => {
+      const deployed = true;
+
+      const { account } = await prepareSpy(deployed);
+
+      const resp = await starknetUtils.getEstimatedFees(
+        STARKNET_SEPOLIA_TESTNET_NETWORK,
+        account.address,
+        account.privateKey,
+        account.publicKey,
+        invocations,
+        transactionVersion,
+      );
+
+      expect(resp).toStrictEqual({
+        suggestedMaxFee: '12',
+        overallFee: '10',
+        unit:
+          transactionVersion === constants.TRANSACTION_VERSION.V2
+            ? FeeTokenUnit.ETH
+            : FeeTokenUnit.STRK,
+        includeDeploy: !deployed,
+      });
+    },
+  );
+
+  it('estimate fees including deployment for transaction version', async () => {
+    const deployed = false;
+    const { account, estimateBulkFeeSpy } = await prepareSpy(deployed);
+
+    const resp = await starknetUtils.getEstimatedFees(
+      STARKNET_SEPOLIA_TESTNET_NETWORK,
+      account.address,
+      account.privateKey,
+      account.publicKey,
+      invocations,
+      '0x2',
+    );
+    const deployAccountpayload = starknetUtils.createAccountDeployPayload(
+      account.address,
+      account.publicKey,
+    );
+
+    expect(estimateBulkFeeSpy).toHaveBeenCalledWith(
+      STARKNET_SEPOLIA_TESTNET_NETWORK,
+      account.address,
+      account.privateKey,
+      [
+        {
+          payload: {
+            addressSalt: account.addressSalt,
+            classHash: ACCOUNT_CLASS_HASH,
+            constructorCalldata: deployAccountpayload.constructorCalldata,
+            contractAddress: deployAccountpayload.contractAddress,
+          },
+          type: 'DEPLOY_ACCOUNT',
+        },
+        {
+          payload: (invocations[1] as any).payload,
+          type: 'INVOKE_FUNCTION',
+        },
+      ],
+      '0x2',
+      undefined,
+    );
+    expect(resp).toStrictEqual({
+      suggestedMaxFee: '12',
+      overallFee: '10',
+      unit: FeeTokenUnit.ETH,
+      includeDeploy: !deployed,
```

Comment:
nit: 
actually i was wonder why we call includeDeploy ^^, should not we just say deploy or not deployed 
we can leave it for now and change it later

Reply:
I agree with you, but lets do later